Eliminate C10_TYPENAME_CONSTEXPR

r-barnes opened the pull request:
Test Plan: Sandcastle

Differential Revision: D64833560

malfet commented on c10/util/TypeIndex.h:
```diff
@@ -76,17 +27,6 @@ struct type_index final : IdWrapper<type_index, uint64_t> {
 
 namespace detail {
 
-#if !defined(__clang__) && !defined(_MSC_VER) && defined(__GNUC__) && \
-    __GNUC__ < 5
-// Getting __PRETTY_FUNCTION__ at compile time only works with GCC >= 5
-#error "You're running a too old version of GCC. We need GCC 5 or later."
-#endif
```
OMG, I though I've cleaned up all of those...

malfet commented on c10/util/TypeIndex.h:
```diff
@@ -11,55 +11,6 @@
-
-// Intel compiler works
-#if defined(__INTEL_COMPILER)
-#define C10_TYPENAME_SUPPORTS_CONSTEXPR 0
-#define C10_TYPENAME_CONSTEXPR
```
Please keep the defines in place, for the sake of backward compatiblity

r-barnes committed:
Eliminate C10_TYPENAME_CONSTEXPR (#138702)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/138702

Test Plan: Sandcastle

Reviewed By: malfet

Differential Revision: D64833560
